### PR TITLE
Add direction to Model.compose()

### PR DESCRIPTION
**Pull request by highfivedenis**
https://github.com/brikteknologier/seraph-model/issues/83

### CODE REVIEW

**Comment by jonpacker** [comment]
Hey! Thanks for taking the initiative. I would love it if there some unit tests for this.

**Comment by highfivedenis** [comment]
i fail to launch test right now, i have to fix that and i will write the test :)

**Comment by deestan** [comment]
Unless I misread the code, I think the new default direction "->" is the opposite of the original direction "<-".  It would be less breaking if the new default matched the old.

**Comment by highfivedenis** [comment]
@deestan you're right i just change it

**Comment by highfivedenis** [comment]
Ready to review

File: lib/read.js [hunk 1]
```diff
@@ -63,8 +63,14 @@ function assembleReadQuery(opts) {
 
     var rels = compositionRelationCypherList(comps);
     var start = level ? '__sm_level' + (level - 1) : opts.varName;
+    var dir = (opts.compMapping[level].matches || opts.compMapping[level][rels.replace(/`/g, '')]).direction;
```
Comment (jonpacker) [inline lib/read.js]: not quite sure I understand what's going on here. where is `matches` coming from?

**Comment by jonpacker** [comment]
Hi! I see there's no tests for actually reading a composited model with different directions on the relationships. It's pretty important, cause there's some code in there that I don't really understand, and I'm not sure if it's working or not without a test to verify.

File: lib/read.js (hunk 1 above)
Comment (highfivedenis) [inline lib/read.js]: i admit it is odd but it's the only manner i found to retrieve the direction.

**Comment by highfivedenis** [comment]
I add a bidirectionnal composition in the test

File: lib/read.js (hunk 1 above)
Comment (jonpacker) [inline lib/read.js]: I'm just not following how it works at all, to be honest. I don't understand where the `matches` property comes from; I don't see how the other side of the `||` will work if there is more than one composited property. Also I can't see how a read would work at all, using this method, if there was one composition with `direction: 'out'`, and one with `direction: 'in'`. I think the best approach is to not specify a direction at all in the read query, and deal with it later on in the `computeAndCoalesce` step (if you even need to deal with it at all? I'm not certain that you need to.)

**Comment by jonpacker** [comment]
Ok, I'm looking for some tests that test a model with both an `out` composition and an `in` composition. There should be one that saves a model with such data, and verifies the integrity of it in the database (as far as I can tell this is all working already); and another one which reads the model back out of the database successfully. It's the reading part which I'm concentrating on here.

**Comment by highfivedenis** [comment]
ready for review

**Comment by highfivedenis** [comment]
up

**Pull request closed without merging**